projects/bms_carrier/src/killswitch.c

```diff
@@ -1,34 +1,35 @@
 #include "killswitch.h"
+
+#include "bms.h"
+#include "exported_enums.h"
 #include "gpio_it.h"
-#include "log.h"
 
 static void prv_killswitch_handler(const GpioAddress *address, void *context) {
-  BatteryHeartbeatStorage *storage = context;
   GpioState state = NUM_GPIO_STATES;
   gpio_get_state(address, &state);
-  if (state == GPIO_STATE_LOW) {
-    // Falling edge - killswitch was hit
-    battery_heartbeat_raise_fault(storage, EE_BATTERY_HEARTBEAT_FAULT_SOURCE_KILLSWITCH);
-  } else {
-    // Rising edge - killswitch was released
-    battery_heartbeat_clear_fault(storage, EE_BATTERY_HEARTBEAT_FAULT_SOURCE_KILLSWITCH);
-  }
+  bool clear = state == GPIO_STATE_LOW ? true : false;
+  fault_bps(EE_BPS_STATE_FAULT_KILLSWITCH, clear);
 }
 
-StatusCode killswitch_init(KillswitchStorage *storage, const GpioAddress *killswitch,
-                           BatteryHeartbeatStorage *battery_heartbeat) {
-  // Force update
-  prv_killswitch_handler(killswitch, battery_heartbeat);
-  return debouncer_init_pin(&storage->debouncer, killswitch, prv_killswitch_handler,
-                            battery_heartbeat);
-}
+StatusCode killswitch_init(DebouncerStorage *storage) {
```

Comment:
Lowkey, I think calling it `_init` is misleading. This module only does one task ... maybe `_setup` or smtg? I'm not sure what we normally do.

Reply:
I think the pattern is to just call anything init. It's initializing gpio pins, so is it really that wrong to call it the init function?

Reply:
lowkey, but it's a nit so up to you. Typically w/ this pattern you init, and then it either does things in the background or you can interact with it. In this case, despite doing inits, it really only does one thing and then exists.